### PR TITLE
Add benchmark and changes for rapidfuzz

### PR DESCRIPTION
**Pull request by pusham7359**
# Related Issue/Addition to code

- Changed calculating Levenshtein distance library from **python-Levenshtein** to **rapidfuzz**.
- This doesn't change the script's logic, it just makes it faster.
- Benchmark (For 10,000 pairs of randomly generated sentences with 10 to 1000 characters):
    - **python-Levenshtein**: 3.83 seconds
    - **rapidfuzz**: 0.12 seconds
- This time difference will be huge with a higher number of sentence pairs.
- **rapidfuzz** is a well-known and widely used library.
- I've added a benchmark file as well for you to compare the time for these libraries. If required I can remove this from PR and just keep the changes for **rapidfuzz**.

## Type of change

- [x] Optimization (non-breaking change which optimizes the code)

# Proposed Changes
- Use rapidfuzz instead of python-Levenshtein

# Additional Info
- Github repo for [rapidfuzz](https://github.com/maxbachmann/RapidFuzz)

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by micziz — APPROVED** [review]
LGTM! Really good performance. (ps, are you paid to say these things?)

**Comment by pusham7359** [comment]
Not at all, just an enthusiast. Checked this repo after Linus' video on this.

**Review by EthanHindmarsh — CHANGES_REQUESTED** [review]
Why include Scripts/benchmark_distance.py?
Other than that, LGTM✅

File: Scripts/benchmark_distance.py
Comment (pusham7359) [reply Scripts/benchmark_distance.py:13]: Scripts/benchmark_distance.py is kept only to validate the performance difference by maintainers/contributors. I can remove the file as it does not add anything to the main logic. Please feel free to advise on this and I'll make changes accordingly.

**Comment by ThioJoe** [comment]
Hmm i'll have to look at this

**Review by TechStudent11 — COMMENTED** [review]
If the scores are what they say on the tin, then I think this improvement will have a great impact.

**Comment by pusham7359** [comment]
@ThioJoe did you get a chance to look at this?

**Comment by ThioJoe** [comment]
I'm holding off on this until I actually get to work on the next feature release 2.17, not sure exactly when I'll get to it

**Comment by DinhHuy2010** [comment]
tested

**Comment by Firecul** [comment]
> tested

Is there a good difference in speed?

**Comment by TechStudent11** [comment]
> tested

Not really useful if you don't provide the results of the test.

**Pull request merged**
